inc/Telemetry/Telemetry.php

```diff
@@ -113,8 +116,16 @@ public function track_remote_data_blocks_usage( int $post_id, WP_Post $post ): v
 			'post_status' => $post_status,
 			'post_type' => $post->post_type,
 		];
-		foreach ( $matches[1] as $match ) {
-			$data_source_type = ConfigStore::get_data_source_type( 'remote-data-blocks/' . $match );
+
+		// This is done this way, to do a one pass processing of the blocks.
+		foreach ( $blocks as $block ) {
```

Comment:
This does not recurse the tree and expects remote data blocks at the top level. It will not find nested blocks such as a remote data block inside a group block.

Reply:
553da9ad

Reply:
Two issues still:

1. Sends a track event even if there are no remote data blocks in the post. We probably want to check if `remote_data_blocks_total_count` is > 0 before sending an event.
1. Minor: Does not find remote data blocks nested in other remote data blocks.

Reply:
06276b0a

Reply:
Thanks for pushing the fix to this @maxschmeling.

I'd like to just port the method we have in the block data api to a library and re-use that, as that would be handy in the future.